cypress/integration/console/shared/overview/overview.spec.js

```diff
@@ -0,0 +1,168 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Overview', () => {
+  const userId = 'main-overview-test-user'
+  const user = {
+    ids: { user_id: userId },
+    primary_email_address: 'view-overview-test-user@example.com',
+    password: 'ABCDefg123!',
+    password_confirm: 'ABCDefg123!',
+  }
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+  })
+
+  describe('Application', () => {
+    const applicationId = 'app-overview-test'
+    const application = {
+      ids: { application_id: applicationId },
+      name: 'Application Test Name',
+      description: `Application Test Description`,
+    }
+
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createApplication(application, userId)
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    beforeEach(() => {
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    })
+
+    it('displays UI elements in place', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/applications/${applicationId}`)
+
+      // Title
+      cy.findByText(`${application.name}`, { selector: 'h1' }).should('be.visible')
+      cy.findByRole('link', { name: /End device/ }).should('be.visible')
+      cy.findByRole('link', { name: /Collaborator/ }).should('be.visible')
+      cy.findByRole('link', { name: /API key/ }).should('be.visible')
+
+      // General Information
+      cy.findByRole('columnheader', { name: 'General information' }).should('be.visible')
```

Comment:
Take care not to be too stiff about display assertions.

I think it's not really relevant for the test case whether this is in a `columnheader` or `cell`. Likewise, I'd suggest not asserting on column header text such as `Application ID` or `Created at`. This is detail for me, meaning that the asserting on the exact text here is irrelevant, similarly to how the user does also not assert to find a text that says exactly `General information` on the overview page, instead he/she asserts to find concrete values of the application somewhere.

It's not always easy to find the right balance with this but please think of ways to avoid us having to refactor tests for minor changes.

Reply:
Also, admittedly, we lack a decision of a clear guideline on how to write such \`displays UI elements in place\` test cases.